packages/world/ts/config/v2/world.test.ts

```diff
@@ -0,0 +1,879 @@
+          },
+          Second: {
+            schema: { secondKey: "Static", secondName: "Dynamic", secondAge: "uint256" },
+            primaryKey: ["secondKey", "secondAge"],
+          },
+        },
+        userTypes: { Static: "address", Dynamic: "string" },
+      });
+      const expected = {
+        tables: {
+          First: {
+            schema: {
+              firstKey: {
+                type: "address",
+                internalType: "Static",
+              },
+              firstName: {
+                type: "string",
+                internalType: "Dynamic",
+              },
+              firstAge: {
+                type: "uint256",
+                internalType: "uint256",
+              },
+            },
+            keySchema: {
+              firstKey: {
+                type: "address",
+                internalType: "Static",
+              },
+              firstAge: {
+                type: "uint256",
+                internalType: "uint256",
+              },
+            },
+            valueSchema: {
+              firstName: {
+                type: "string",
+                internalType: "Dynamic",
+              },
+            },
+            primaryKey: ["firstKey", "firstAge"],
+          },
+          Second: {
+            schema: {
+              secondKey: {
+                type: "address",
+                internalType: "Static",
+              },
+              secondName: {
+                type: "string",
+                internalType: "Dynamic",
+              },
+              secondAge: {
+                type: "uint256",
+                internalType: "uint256",
+              },
+            },
+            keySchema: {
+              secondKey: {
+                type: "address",
+                internalType: "Static",
+              },
+              secondAge: {
+                type: "uint256",
+                internalType: "uint256",
+              },
+            },
+            valueSchema: {
+              secondName: {
+                type: "string",
+                internalType: "Dynamic",
+              },
+            },
+            primaryKey: ["secondKey", "secondAge"],
+          },
+        },
+        userTypes: { Static: "address", Dynamic: "string" },
+        enums: {},
+        namespace: "",
+        namespaces: {},
+      } as const;
+      attest<typeof expected>(config);
+    });
+
+    it("should throw if referring to fields of different tables", () => {
+      attest(
+        resolveWorldConfig({
+          tables: {
+            First: {
+              schema: { firstKey: "address", firstName: "string", firstAge: "uint256" },
+              primaryKey: ["firstKey", "firstAge"],
+            },
+            Second: {
+              schema: { secondKey: "address", secondName: "string", secondAge: "uint256" },
+              // @ts-expect-error Type '"firstKey"' is not assignable to type '"secondKey" | "secondAge"'
```

Comment:
not at all blocking, but where this condition lives, do we have room to expand on/improve this error type?

like a branch that outputs one of the following:
> `firstKey` field was not found in the schema
> `firstKey` field is not a static ABI type

Reply:
this one is a native typescript error, it's probably possible to create a validator function that routes this condition to a custom string error, but would leave that for later